File: testdata/fixtures/ruby/CWE-327.rb
# CWE-327 Use of Broken or Risky Cryptographic Algorithm in Ruby
# MD5 and SHA1 are broken for collision resistance; SHA-256/SHA-512 should be used.

require 'digest'
require 'openssl'

def hash_data(data)
  # ruleid: ruby-insecure-hash
  Digest::MD5.hexdigest(data)

  # ruleid: ruby-insecure-hash
  Digest::SHA1.hexdigest(data)

  # ruleid: ruby-insecure-hash
  Digest::MD5.new

  # ruleid: ruby-insecure-hash
  OpenSSL::Digest::MD5.hexdigest(data)
end

def safe(data)
  # ok: ruby-insecure-hash
  Digest::SHA256.hexdigest(data)

  # ok: ruby-insecure-hash
  Digest::SHA512.hexdigest(data)

  # ok: ruby-insecure-hash
  OpenSSL::Digest::SHA256.hexdigest(data)
end

File: testdata/fixtures/rust/CWE-327.rs
// CWE-327: insecure hashes — MD5, SHA1 etc are cryptographically broken
use md5::Md5;
use sha1::Sha1;
use sha2::Sha256;

fn main() {
    // ruleid: insecure-hashes
    let _ = md5::Md5::new();
    // ruleid: insecure-hashes
    let _ = sha1::Sha1::new();
    // ruleid: insecure-hashes
    let _ = md5::Md5::new();
    // ok: insecure-hashes
    let _ = Sha256::new();
    // ok: insecure-hashes
    let _ = sha2::Sha256::new();
    // ok: insecure-hashes
    let _ = 42;
}
